python/TestHarness/CSVDiffer.py

```diff
@@ -89,6 +89,14 @@ def diff(self):
           if abs(val2) < abs_zero:
             val2 = 0
 
+          # disallow nan in the gold file
+          if math.isnan(val1) or math.isnan(val2):
+            self.addError(fname, "The values in column \"" + key.strip() + "\" is nan")
```

Comment:
1. The comment seems to be misleading. You are not allowing `NaN`s in both files. Typically the first one is the gold file (IIRC)
1. "Values is a NaN" is not correct English. You can say "Column XYZ contains NaN".

Similarly below...